api/websocket/server/server.go

```diff
@@ -570,3 +593,17 @@ func (ws *WsServer) sendInboundRelayMessageToClient(v interface{}) {
 		log.Error("Decode relay message failed")
 	}
 }
+
+// client auth, generate challenge
+func (ws *WsServer) sendClientAuthChallenge(sess *session.Session) {
+	resp := make(map[string]interface{})
+	resp["Action"] = "authChallenge"
+
+	var challenge [64]byte
+	rand.Reader.Read(challenge[:])
+	resp["Challenge"] = hex.EncodeToString(challenge[:])
+	sess.Challenge = challenge
+	resp["Error"] = errcode.SUCCESS
+
+	ws.respondToSession(sess, resp)
```

Comment:
It would be good to change `respondToSession` and make it return error. Same for `sendClientAuthChallenge`

Reply:
fixed